packages/amplify_auth_cognito/ios/Classes/FlutterUpdateUserAttributeResult.swift

```diff
@@ -18,99 +18,18 @@ import Foundation
-        ]
-      ]
-    }
-}
-
-func isComplete(res: AmplifyOperation<AuthUpdateUserAttributeRequest, AuthUpdateAttributeResult, AuthError>.OperationResult) -> Bool {
-  var complete: Bool = false;
-  switch res {
-    case .success(let signUpResult):
-      complete = signUpResult.isUpdated
-    case .failure:
-      complete = false
-  }
-  return complete;
-}
-
-func setCodeDeliveryDetails(res: AmplifyOperation<AuthUpdateUserAttributeRequest, AuthUpdateAttributeResult, AuthError>.OperationResult) -> [String: String] {
-    var deliveryMap: [String: String] = [:]
-    switch res {
-        case .success(let updateResult):
-          if case let .confirmAttributeWithCode(deliveryDetails, _) = updateResult.nextStep {
-            if case let .email(e) = deliveryDetails.destination {
-              deliveryMap["destination"] = e! as String
-              deliveryMap["attributeName"] = "email"
-              deliveryMap["deliveryMedium"] = "EMAIL"
-            }
-
-            if case let .phone(e) = deliveryDetails.destination {
-              deliveryMap["destination"] = e! as String
-              deliveryMap["attributeName"] = "phone"
-            }
-
-            if case let .sms(e) = deliveryDetails.destination {
-              deliveryMap["destination"] = e! as String
-              deliveryMap["attributeName"] = "sms"
-              deliveryMap["deliveryMedium"] = "SMS"
-            }
-
-            if case let .unknown(e) = deliveryDetails.destination {
-              deliveryMap["destination"] = e! as String
-              deliveryMap["attributeName"] = "unknown"
-            }
-          }
+        switch res {
+        case .success(let res):
+            self.result = res
         case .failure:
-            deliveryMap = [:]
+            self.result = nil
+        }
     }
-    return deliveryMap
-}
 
-func setAdditionalInfo(res:  AmplifyOperation<AuthUpdateUserAttributeRequest, AuthUpdateAttributeResult, AuthError>.OperationResult) -> [String: String] {
-    var infoMap: [String: String] = [:]
-    switch res {
-        case .success(let updateResult):
-          if case let .confirmAttributeWithCode(_, additionalInfo) = updateResult.nextStep {
-            infoMap = additionalInfo ?? [:]
-          }
-        case .failure:
-            infoMap = [:]
-    }
-    return infoMap
-}
-
-func setState(res: AmplifyOperation<AuthUpdateUserAttributeRequest, AuthUpdateAttributeResult, AuthError>.OperationResult) -> String {
-    let state: String = "ERROR"
-    switch res {
-        case .success(let updateResult):
-          if case .done = updateResult.nextStep {
-             return  "DONE"
-          }
-          if case .confirmAttributeWithCode = updateResult.nextStep {
-            return "CONFIRM_ATTRIBUTE_WITH_CODE"
-          }
-        case .failure:
-           return "ERROR"
+    func toJSON() -> Dictionary<String, Any> {
+        return self.result == nil ? [:] : serializeAuthUpdateAttributeResult(result: self.result!)
```

Comment:
Nit suggestion/question: What do you think about moving the nil check into the `serializeAuthUpdateAttributeResult` function itself and have it return the empty dict on a nil result instead? It seems like having the nil guard there would arguably make it a bit more reusable.

Reply:
I think that is reasonable. I put it here because I viewed it as specific to this use case. In this case, result is only nil if the AmplifyOperation is a failure, and it doesn't really matter what this returns in the case of a failure.

It is admittedly a bit hacky though to just return an empty dictionary in the case of a failure, knowing it won't be used. I don't think there is a clear "correct" return value in the case of a failure, but an empty dict probably isn't appropriate. Looking at what we do for the other results in auth, it looks like we generally return a dictionary that has the same attributes as the success case with all the values as empty string / nil / false. Again, not sure this is clearly "correct" but consistency would be preferred.

tl;dr - I will update serializeAuthUpdateAttributeResult to allow for nil and follow the pattern used elsewhere for serializing results in the case of failures 👍